app/src/main/java/com/goryachok/forecastapp/view/activity/MainActivity.kt

```diff
@@ -4,41 +4,92 @@ import android.app.SearchManager
 import android.os.Bundle
 import android.view.Menu
 import android.view.MenuItem
+import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.SearchView
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.gms.location.LocationServices
+import com.goryachok.forecastapp.ConnectivityListener
 import com.goryachok.forecastapp.R
 import com.goryachok.forecastapp.view.FragmentsAdapter
 import com.goryachok.forecastapp.view.fragment.CurrentFragment
+import com.goryachok.forecastapp.view.fragment.DailyFragment
+import com.goryachok.forecastapp.view.fragment.HourlyFragment
+import com.goryachok.forecastapp.view.fragment.MyFragment
+import com.goryachok.forecastapp.viewmodel.MainViewModel
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
+    private val viewModelFactory by lazy {
+        object : ViewModelProvider.Factory {
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+                return MainViewModel() as T
+            }
+        }
+    }
+
+    val viewModel by lazy {
+        ViewModelProvider(
+            this,
+            viewModelFactory
+        ).get(MainViewModel::class.java)
+    }
+
+    private val connectivityListener by lazy {
+        ConnectivityListener(
```

Comment:
Please refactor this, that ConnectivityListener has 2 callbacks but not show/hide snackbar functions.

Reply:
Done